include/mem.h

```diff
@@ -41,136 +41,172 @@ void MEM_A20_Enable(bool enable);
+ * physical addressing */
 
-static INLINE Bit8u real_readb(Bit16u seg,Bit16u off) {
-	return mem_readb((seg<<4)+off);
+static inline uint8_t real_readb(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return mem_readb(base + off);
 }
-static INLINE Bit16u real_readw(Bit16u seg,Bit16u off) {
-	return mem_readw((seg<<4)+off);
+static inline uint16_t real_readw(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return mem_readw(base + off);
 }
-static INLINE Bit32u real_readd(Bit16u seg,Bit16u off) {
-	return mem_readd((seg<<4)+off);
+static inline uint32_t real_readd(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return mem_readd(base + off);
 }
 
-static INLINE void real_writeb(Bit16u seg,Bit16u off,Bit8u val) {
-	mem_writeb(((seg<<4)+off),val);
+static inline void real_writeb(uint16_t seg, uint16_t off, uint8_t val)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	mem_writeb(base + off, val);
 }
-static INLINE void real_writew(Bit16u seg,Bit16u off,Bit16u val) {
-	mem_writew(((seg<<4)+off),val);
+static inline void real_writew(uint16_t seg, uint16_t off, uint16_t val)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	mem_writew(base + off, val);
 }
-static INLINE void real_writed(Bit16u seg,Bit16u off,Bit32u val) {
-	mem_writed(((seg<<4)+off),val);
+static inline void real_writed(uint16_t seg, uint16_t off, uint32_t val)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	mem_writed(base + off, val);
 }
 
-
-static INLINE Bit16u RealSeg(RealPt pt) {
-	return (Bit16u)(pt>>16);
+static inline uint16_t RealSeg(RealPt pt)
+{
+	return (uint16_t)(pt >> 16);
 }
 
-static INLINE Bit16u RealOff(RealPt pt) {
-	return (Bit16u)(pt&0xffff);
+static inline uint16_t RealOff(RealPt pt)
+{
+	return (uint16_t)(pt & 0xffff);
 }
 
-static INLINE PhysPt Real2Phys(RealPt pt) {
-	return (RealSeg(pt)<<4) +RealOff(pt);
+static inline PhysPt Real2Phys(RealPt pt)
+{
+	const auto base = static_cast<uint32_t>(RealSeg(pt) << 4);
+	return base + RealOff(pt);
 }
 
-static INLINE PhysPt PhysMake(Bit16u seg,Bit16u off) {
-	return (seg<<4)+off;
+static inline PhysPt PhysMake(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return base + off;
 }
 
-static INLINE RealPt RealMake(Bit16u seg,Bit16u off) {
-	return (seg<<16)+off;
+static inline RealPt RealMake(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 16);
+	return base + off;
 }
 
-static INLINE void RealSetVec(Bit8u vec,RealPt pt) {
-	mem_writed(vec<<2,pt);
+static inline void RealSetVec(uint8_t vec, RealPt pt)
+{
+	const auto target = static_cast<uint16_t>(vec << 2);
+	mem_writed(target, pt);
 }
 
-static INLINE void RealSetVec(Bit8u vec,RealPt pt,RealPt &old) {
-	old = mem_readd(vec<<2);
-	mem_writed(vec<<2,pt);
+static inline void RealSetVec(uint8_t vec, RealPt pt, RealPt &old)
```

Comment:
I think this function is not really necessary; add a `TODO` comment for removing it.

I looked up all 3 places where this function is used and in all cases the 3rd parameter is updated then ignored.

Reply:
Flagged it.